lib/matplotlib/tests/test_axes.py

```diff
@@ -4593,13 +4593,13 @@ def test_margins():
 
     fig2, ax2 = plt.subplots(1, 1)
     ax2.plot(data)
-    ax2.margins(1, 0.5)
-    assert ax2.margins() == (1, 0.5)
+    ax2.margins(0.5, 2)
+    assert ax2.margins() == (0.5, 2)
 
     fig3, ax3 = plt.subplots(1, 1)
     ax3.plot(data)
-    ax3.margins(x=1, y=0.5)
-    assert ax3.margins() == (1, 0.5)
+    ax3.margins(x=-0.2, y=0.5)
+    assert ax3.margins() == (-0.2, 0.5)
```

Comment:
I think these tests would be better if they also asserted the x-lim and y-lim you expect.  i.e. `assert ax3.get_xlim() == (left, right)`.  That way if anyone monkeys with the internal code this test will catch the inconsistency.

Reply:
Pushed 76a2721.